app/src/components/blocks/selectStage/SelectStage.tsx

```diff
@@ -39,18 +41,54 @@ function SelectStage({ uploadDocuments, setDocuments, patientDetails }: Props) {
                 isFile: () => {
                     return !!hasFileInput || 'Please select a file';
                 },
-                isLessThan5GB: (value?: Array<UploadDocument>) => {
+                perFileValidation: (value?: Array<UploadDocument>) => {
                     if (Array.isArray(value)) {
                         for (let i = 0; i < value.length; i++) {
-                            if (value[i].file.size > FIVEGB) {
+                            const currentFile = value[i].file;
+                            if (currentFile.size > FIVEGB) {
                                 return 'Please ensure that all files are less than 5GB in size';
                             }
+                            if (name === 'lg-documents') {
+                                if (currentFile.type !== 'application/pdf') {
+                                    return 'One or more of the files do not match the required file type. Please check the file(s) and try again';
+                                }
+                                const expectedNumberOfFiles = currentFile.name.match(lgFilesNumber);
+                                if (
+                                    !lgRegex.exec(currentFile.name) ||
+                                    (!!expectedNumberOfFiles &&
+                                        (value.length !==
+                                            parseInt(expectedNumberOfFiles[0].slice(2)) ||
+                                            parseInt(currentFile.name.split(lgFilesNumber)[0]) >
+                                                parseInt(expectedNumberOfFiles[0].slice(2)))) ||
+                                    currentFile.name.split(lgFilesNumber)[1] !==
+                                        value[0].file.name.split(lgFilesNumber)[1] ||
+                                    currentFile.name.split(lgFilesNumber)[0] === '0'
+                                ) {
```

Comment:
This operator is unreadable. Please break down the boolean values into their own named constants then compare the constants

const hasNoFiles = 
const doesPassRegex = 
const doesPassValidation = hasNofiles && doesPassRegex
if(doesPassValidation)

for example

Reply:
done